### ISSUE TITLE
bug: props unrelated to HTMLElement are being flagged by `reserved-member-names` rule

### ISSUE THREAD

**Issue by jcfranco** [stencil-community/stencil-eslint#100]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/stencil-community/stencil-eslint/issues) that already report this problem, without success.

### Stencil ESLint Version

0.7.1

### Current Behavior

https://github.com/stencil-community/stencil-eslint/pull/67/files introduced properties unrelated to `HTMLElement` that are being flagged. 

For example, a `color` prop will produce a warning/error despite it not being defined in [`HTMLElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement):

>   126:28  error    The @Prop name "color conflicts with a key in the HTMLElement prototype. Please choose a different name  @stencil-community/reserved-member-names

### Expected Behavior

Only attributes/properties on `HTMLElement.prototype` to be flagged by the rule. 

### Steps to Reproduce

The following should produce the error described above:

```tsx
import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'my-component',
  shadow: true,
})
export class MyComponent {
  @Prop() color: string;

  render() {
    return <div>color is {this.color}</div>;
  }
}
```

### Code Reproduction URL

https://github.com/ionic-team/stencil-component-starter

**Comment by jcfranco** [comment]
I shamelessly took a shortcut with the repro URL and steps, but please LMK if this is still needed and I'll make the appropriate changes.

**Comment by raphaelpor** [comment]
Hello @jcfranco,
Thanks for reporting that. But I checked again and `color` is an expected HTML attribute. Specifically used in the `<font>` element (Ref.: https://www.geeksforgeeks.org/html-font-color-attribute/).

I understand it is not a common practice, but since we need to complain with the HTML standards, I would recommend keeping it there.

What do you think? :)

**Comment by raphaelpor** [comment]
You are right, @jcfranco.

Thanks for the patience to explain it. I will create a PR removing that lock for `color`, soon. 🙏

**Comment by jcfranco** [comment]
@raphaelpor Hope you don't mind, but I [took a stab](https://github.com/stencil-community/stencil-eslint/pull/101) at this. 😅

**Comment by jcfranco** [comment]
@raphaelpor Friendly ping. 😁

**Comment by raphaelpor** [comment]
> @raphaelpor Friendly ping. 😁

Thanks for reminding me @jcfranco. I will make sure to have a look at that, this week. 🙏

**Comment by James-Wilkinson-git** [comment]
If you use svelte any HTML elements attribute is reserved and it will yell at you when inserting a web component with any of these https://www.w3schools.com/tags/ref_attributes.asp

**Comment by jcfranco** [comment]
@James-Wilkinson-git I _think_ that's outside of the scope of this rule. Custom elements extend `HTMLElement`, so reserved names should be in the context of that class. Frameworks warning against attributes/properties from elements extending `HTMLElement` is a separate issue unrelated to Stencil.

**Comment by James-Wilkinson-git** [comment]
It’s a best practice to not use any reserved html element not just global ones, as your web component should work everywhere, if you use a reserved attribute fire another element this breaks and there should be a rule to guard against it.

**Comment by James-Wilkinson-git** [comment]
I think a better AirTag that is more backwards compatible is to create two rules. One for global attributes and one for all attributes. All attributes is needed for anyone who’s consumer acts like svelte and manes assumptions on how to handle each HTML attribute for binding.
